### PR TITLE
Filter bad data (code 0) from the datalake.

### LINKED ISSUE #365: Filter "bad" data (code 0) from the datalake
The csv files we read here in the Equinor has a status column, which is either 192 (Good) or 0 (Bad).

We should filter out bad data on reading.

### PR DESCRIPTION
**Pull request by EinarJohnsen**
Removes rows with code 0 from the data in the ncs_reader. Fixes #365

### CODE REVIEW

**Review by epa095 — COMMENTED** [review]
Definitely looking good. But do you remove the "status" column again, or is also returned now?

File: gordo_components/data_provider/ncs_reader.py
Comment (milesgranger) [inline gordo_components/data_provider/ncs_reader.py]: As we have lots of 'Status codes', perhaps we should change this to `filter_status_codes = [0]` and this way we can support any future changes in status codes or other requirements without hard-coding.

File: gordo_components/data_provider/ncs_reader.py
Comment (milesgranger) [inline gordo_components/data_provider/ncs_reader.py]: And then here, it would be a one liner; `df = df[df["Status"].isin(filter_status_codes)]`

File: tests/gordo_components/data_provider/test_ncs_reader.py
Comment (milesgranger) [inline tests/gordo_components/data_provider/test_ncs_reader.py]: This isn't a frame, but rather a series generator, no?

File: tests/gordo_components/data_provider/test_ncs_reader.py
Comment (milesgranger) [inline tests/gordo_components/data_provider/test_ncs_reader.py]: Can we instead do something like this:
```python
n_expected = 15 if remove_bad_values else 20
assert all(len(series) == n_expected for series in series_gen)
```

File: tests/gordo_components/data_provider/test_ncs_reader.py
Comment (EinarJohnsen) [reply tests/gordo_components/data_provider/test_ncs_reader.py]: This is true, it is a series generator.

File: gordo_components/data_provider/ncs_reader.py
Comment (EinarJohnsen) [reply gordo_components/data_provider/ncs_reader.py]: Would ```remove_status_codes = [0]``` be a better name? ```filter_status_code``` is kind of ambiguous, you have to look at the code to see if the status codes are kept or removed. @milesgranger

File: gordo_components/data_provider/ncs_reader.py
Comment (milesgranger) [inline gordo_components/data_provider/ncs_reader.py]: Will need to update the parameter docs and its type.

**Review by milesgranger — COMMENTED** [review]
At least from my end, just had one last comment. After that, you've got my approval I think.

**Review by milesgranger — APPROVED** [review]
:100: :tophat: :100:

**Pull request merged**
